bin/posthog-worktree

```diff
@@ -39,6 +39,31 @@ done
 # Worktree base directory (configurable via POSTHOG_WORKTREE_BASE)
 WORKTREE_BASE="${POSTHOG_WORKTREE_BASE:-${HOME}/.worktrees/posthog}"
 
+# Copy helper that uses copy-on-write clones when available (APFS/Btrfs)
+# Returns 0 on success, 1 if CoW not available and require_cow is true
+clone_path() {
+    local source_path="$1"
+    local destination_path="$2"
+    local require_cow="${3:-false}"  # If true, skip copy when CoW unavailable
+    local cp_args=("-a")
+
+    if [[ "$(uname -s)" == "Darwin" ]]; then
+        if cp -c "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null; then
+            return 0
+        fi
+    else
+        if cp --reflink=auto "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null; then
+            return 0
+        fi
```

Comment:
`--reflink=auto` silently falls back to regular copy on non-CoW filesystems (ext4), so this always succeeds and returns 0. The `require_cow=true` logic below never triggers on Linux.
 
To actually require CoW, use `--reflink=always` when `require_cow` is `true`:

```bash
if [[ "$require_cow" == "true" ]]; then
    cp --reflink=always "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null
else
    cp --reflink=auto "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null
fi
```

Otherwise Linux users on ext4 get silent 4GB copies instead of the intended skip.